airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/oauth.py

```diff
@@ -80,3 +82,109 @@ def access_token(self) -> str:
+            self.get_refresh_token(),
+            scopes,
+            token_expiry_date,
+            access_token_name,
+            expires_in_name,
+            refresh_request_body,
+            grant_type,
+        )
+
+    def _validate_connector_config(self):
+        """Validates the defined getters for configuration values are returning values.
+
+        Raises:
+            ValueError: Raised if the defined getters are not returning a value.
+        """
+        for field_name, getter in [
+            ("client_id", self.get_client_id),
+            ("client_secret", self.get_client_secret),
+            (self.get_refresh_token_name(), self.get_refresh_token),
+        ]:
+            try:
+                assert getter()
+            except (AssertionError, KeyError):
+                raise ValueError(
+                    f"This authenticator expects a {field_name} field under the {self.credentials_configuration_field_name} field. Please override this class getters or change your configuration structure."
+                )
+
+    def get_refresh_token_name(self) -> str:
+        return self._refresh_token_name
+
+    def _get_config_credentials_field(self, field_name):
+        return self._connector_config[self.credentials_configuration_field_name][field_name]
+
+    def get_client_id(self) -> str:
+        return self._get_config_credentials_field("client_id")
+
+    def get_client_secret(self) -> str:
+        return self._get_config_credentials_field("client_secret")
+
+    def set_refresh_token(self, new_refresh_token: str):
+        """Set the new refresh token value. The mutation of the connector_config object will emit an Airbyte control message.
+
+        Args:
+            new_refresh_token (str): The new refresh token value.
+        """
+        self._connector_config[self.credentials_configuration_field_name][self.get_refresh_token_name()] = new_refresh_token
+
+    def get_refresh_token(self) -> str:
+        return self._get_config_credentials_field(self.get_refresh_token_name())
+
+    def get_access_token(self) -> str:
+        """Retrieve new access and refresh token if the access token has expired.
+        The new refresh token is persisted with the set_refresh_token function
+        Returns:
+            str: The current access_token, updated if it was previously expired.
+        """
+        if self.token_has_expired():
+            t0 = pendulum.now()
+            new_access_token, access_token_expires_in, new_refresh_token = self.refresh_access_token()
+            self.access_token = new_access_token
+            self.set_token_expiry_date(t0.add(seconds=access_token_expires_in))
+            self.set_refresh_token(new_refresh_token)
+        return self.access_token
+
+    def refresh_access_token(self) -> Tuple[str, int, str]:
+        try:
+            response_json = self._get_refresh_access_token_response()
+            return (
+                response_json[self.get_access_token_name()],
```

Comment:
would recommend using `dpath`

Reply:
Agreed but for implementation consistency with parent class and abstract class, I'd prefer to do this in a separate PR in which we can replace the `access_token_name`, `expires_in_name`, and `refresh_token_name` by dpaths that can be used when parsing these responses. Wdyt?

Reply:
makes sens!